### ISSUE TITLE
Expression will always be true

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#503]
It is hard to validate on PHP, but in some cases it is very clear.

```php
function example(User $user) {
    if ($user instanceof User) { }
    if ($user instanceof SuperUser) { }
}
```

In this case, both `if()` will always be true (supposing that `SuperUser extensd User`). 

The reverse condition could be checked too:

```php
function example(User $user) {
    if ($user instanceof NotUser) { }
}
```

Shoud display "_this expression will be always false_".

Note: surprising, `if(true)` doesn't reporting anything currently. It should be fixed by this inspection too.

**Comment by kalessil** [comment]
Covered by #229
